bpo-33609: small fix to wording in help file

nascheme opened the pull request:
Improve the wording about the ordering of dict keys.

<!-- issue-number: bpo-33609 -->
https://bugs.python.org/issue33609
<!-- /issue-number -->

nascheme committed:
Update stdtypes.rst

ned-deily commented on Doc/library/stdtypes.rst:
```diff
@@ -4248,7 +4248,7 @@ pairs within braces, for example: ``{'jack': 4098, 'sjoerd': 4127}`` or ``{4098:
-   Dict preserves insertion order.  Note that updating key doesn't affects the
+   Dict preserves insertion order.  Note that updating a key does not affect the
    order.  On the other hand, keys added after deletion are inserted to the
    last. ::
```
While you're in there: "inserted to the last" -> "inserted at the end" ?  Also, I notice that this most recent change uses "dict" whereas the rest of the section uses "dictionary".

nascheme committed:
More wording tweaks for dict ordering docs.

Say "Dictionary" rather than "Dict" (matches rest of document).  Say "inserted at the end" rather than "inserted to the last".
